### PR TITLE
[WIP] Feature/exit survey

### PR DESCRIPTION
**Pull request by leb2dg**
### In Progress
* validate all fields on submit not just ones that have been accessed

### Current State
screen shot 2016-03-01 at 1 33 25 pm
screen shot 2016-03-01 at 1 39 14 pm
screen shot 2016-03-01 at 1 33 05 pm

### CODE REVIEW

File: exp-player/addon/components/exp-exit-survey.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+import Ember from 'ember';
+
+import ExpFrameBaseComponent from 'exp-player/components/exp-frame-base';
+import layout from '../templates/components/exp-exit-survey';
+
+const defaultSchema = {
+    "schema": {
```
Comment (samchrisinger) [inline exp-player/addon/components/exp-exit-survey.js]: Minor stylistic thing: quotes aren't really necessary on the key values here

File: exp-player/addon/components/exp-exit-survey.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                id: {
+                    type: 'string',
+                    description: 'A unique identifier for this item'
+                },
+                title: {
+                    type: 'string',
+                    description: 'A title for this item',
+                    default: 'Exit Survey'
+                },
+                exitThankYou: {
```
Comment (samchrisinger) [inline exp-player/addon/components/exp-exit-survey.js]: Do you expect this attributes will be passed in from the data source (i.e. the experiment.json)? If not they can just be regular attributes of the component.

File: exp-player/addon/components/exp-exit-survey.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        return {
+            update: function () {
+                this.refreshValidationState(true);
+                if (this.isValid(true)) {
+                   root.set('formData', this.getValue());
+                   root.set('section1', false);
+                }
+            },
+            finish: function() {
+                root.set('additionalFormData', this.getValue());
```
Comment (samchrisinger) [inline exp-player/addon/components/exp-exit-survey.js]: Could do ```javascript
var formData = root.get('formData');
Ember.merge(formData, this.getValue());
root.set('formData', formData);
```

File: exp-player/addon/components/exp-exit-survey.js (hunk 2 above)
Comment (leb2dg) [inline exp-player/addon/components/exp-exit-survey.js]: All of them except for exitThankYou and exitMessage will come from the context that is not yet available (this PR can wait until it is though). The exitThankYou/exitMessage could be in the template, an attribute of the component, or passed in. Is there a preference?

Comment (samchrisinger) [inline exp-player/addon/components/exp-exit-survey.js]: Let's go ahead and pass them in.

**Pull request merged**